src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -0,0 +1,458 @@
+
+        # TODO: check if Double-checking using yum/zypper ps equivalent should be used?
+        # TODO: check what this command indicates: tdnf update --assumeno --reboot-required
+        return False
+    # endregion
+
+    # region To review and remove later
+
+    def get_current_auto_os_patch_state(self):
+        """ Gets the current auto OS update patch state on the machine """
+        pass
+
+    def disable_auto_os_update(self):
+        """ Disables auto OS updates on the machine only if they are enabled and logs the default settings the machine comes with """
+        pass
+
+    def backup_image_default_patch_configuration_if_not_exists(self):
+        """ Records the default system settings for auto OS updates within patch extension artifacts for future reference.
+        We only log the default system settings a VM comes with, any subsequent updates will not be recorded"""
+        pass
+
+    def is_image_default_patch_configuration_backup_valid(self, image_default_patch_configuration_backup):
+        pass
+
+    def update_os_patch_configuration_sub_setting(self, patch_configuration_sub_setting, value, patch_configuration_sub_setting_pattern_to_match):
+        pass
+
+    def add_arch_dependencies(self, package_manager, package, version, packages, package_versions, package_and_dependencies, package_and_dependency_versions):
+        """
+        Add the packages with same name as that of input parameter package but with different architectures from packages list to the list package_and_dependencies.
+        Parameters:
+        package_manager (PackageManager): Package manager used.
+        package (string): Input package for which same package name but different architecture need to be added in the list package_and_dependencies.
+        version (string): version of the package.
+        packages (List of strings): List of all packages selected by user to install.
+        package_versions (List of strings): Versions of packages in packages list.
+        package_and_dependencies (List of strings): List of packages along with dependencies. This function adds packages with same name as input parameter package
+                                                    but different architecture in this list.
+        package_and_dependency_versions (List of strings): Versions of packages in package_and_dependencies.
+        """
+        package_name_without_arch = package_manager.get_product_name_without_arch(package)
+        for possible_arch_dependency, possible_arch_dependency_version in zip(packages, package_versions):
+            if package_manager.get_product_name_without_arch(possible_arch_dependency) == package_name_without_arch and possible_arch_dependency not in package_and_dependencies and possible_arch_dependency_version == version:
+                package_and_dependencies.append(possible_arch_dependency)
+                package_and_dependency_versions.append(possible_arch_dependency_version)
```

Comment:
Please create a test case matrix of scenarios and install patch input - in that doc, you can point to a placeholder for having all this tested (and output marked there) to work towards resolve this comment.

Reply:
Pending

Reply:
done